package/MDAnalysis/coordinates/base.py

```diff
@@ -1282,8 +1308,85 @@ def __repr__(self):
                     nframes=self.n_frames,
                     natoms=self.n_atoms
                 ))
+                
+    def add_auxiliary(self, auxname, auxdata, **kwargs):
+        """Add auxiliary data to be read alongside trajectory.
+
+        Auxiliary data may be any data timeseries from the trajectory additional
+        to that read in by the trajectory reader. *auxdata* can be an AuxReader 
+        instance, or the data itself as e.g. an array, filename; in the latter 
+        case an appropriate AuxReader is guessed from the data/file format.
+
+        The representative value of the auxiliary data for each timestep (as
+        calculated by the AuxReader) is stored in the ts.aux namespace as 
+        auxname.
+
+        e.g. to add additional pull force data stored in pull-force.xvg::
 
+          u = MDAnalysis.Universe(PDB, XTC)
+          u.trajectory.add_auxiliary('pull', 'pull-force.xvg')
 
+        The representative value for the current timestep may then be accessed 
+        as ``u.trajectory.ts.aux.pull``.
+
+        AuxReader is initially matched to the current timestep of the trajectory.
+
+        """
+        if auxname in self.aux_list:
+            raise ValueError("Auxiliary data with name {name} already "
+                             "exists".format(name=auxname))
+        if isinstance(auxdata, AuxReader):
+            aux = auxdata
+        else:
+            auxreader = get_auxreader_for(auxdata)
+            aux = auxreader(auxname, auxdata, **kwargs)
+        self._auxs[auxname] = aux
+        self.ts = aux.go_to_ts(self.ts)
+    
+    def remove_auxiliary(self, auxname):
+        """Close the Reader for auxiliary data *auxname* and remove data 
+        from trajectory."""
+        if auxname in self.aux_list:
+            self._auxs[auxname].close()            
+            del self._auxs[auxname]
+            del self.ts.aux.__dict__[auxname]
```

Comment:
You shouldn't really be setting stuff in this way... it looks strange to me?  Maybe better would be to make the `Namespace` object smarter by just hacking a `dict` to allow attribute/lazy style access

```python
class Namespace(dict):
    def __getattr__(self, key):
        # a.this causes a __getattr__ call for key = 'this' 
        return self[key]
    
    def __setattr__(self, key, value):
        # a.this = 10 causes a __setattr__ call for key='this' value=10
        self[key] = value
```